system/jlib/jstats.cpp

```diff
@@ -1988,9 +2012,69 @@ class CStatisticCollection : public CInterfaceOf<IStatisticCollection>
             cur.visit(visitor);
     }
 
+    virtual bool refreshAggregates(std::vector<StatisticKind> & aggregateKinds, std::vector<unsigned __int64> & totals) override
+    {
+        assertex(aggregateKinds.size()==totals.size());
+
+        bool updated = false;
+        if (queryScopeType()==SSTsubgraph)
+        {
+            ForEachItemIn(i, stats)
+            {
+                Statistic & stat = stats.element(i);
+                StatisticKind kind = stat.queryKind();
+                auto iteratorVec = std::find(aggregateKinds.begin(), aggregateKinds.end(), kind);
```

Comment:
Are you sure that the order of the statistics is consistent? For example, if a statistic has a value of zero, addStatistic/updateStatistic may not be called so it would not be in stats.